feat(firmware): use new overlay-based configuration

coderbyheart opened the pull request:
See https://github.com/nrfconnect/sdk-nrf/pull/5662

BREAKING CHANGE: firmware build step has significantly changed

coderbyheart committed:
feat(firmware): use new overlay-based configuration

See https://github.com/nrfconnect/sdk-nrf/pull/5662

BREAKING CHANGE: firmware build step has significantly changed

wiba-nordic commented on docs/aws/GettingStarted/FirmwareConfiguration.rst:
```diff
@@ -11,52 +11,12 @@ To configure the :ref:`firmware <firmware-aws-index>`, complete the following ma
       node cli info -o mqttEndpoint
 
-#. Use this MQTT endpoint value as the value for the ``CONFIG_AWS_IOT_BROKER_HOST_NAME`` configuration variable.
+#. Add the setting to a file called :file:`firmware.conf`:
```
```suggestion
#. Add the ``CONFIG_AWS_IOT_BROKER_HOST_NAME`` setting to the :file:`firmware.conf` file, using the endpoint value from the previous step as the value:
```
This assumes that `firmware.conf` already exists by default. If it does not, it should have a step dedicated to creating it, probably as the first step.

coderbyheart committed:
fix: create firmware.conf file

... and explain in text which setting to add.

Co-Auhored-By: Wille Backmann <wille.backman@nordicsemi.no>
